src/vcTiffConvert.cpp

```diff
@@ -10,157 +10,592 @@
 #include "udMath.h"
 #include "tiffio.h"
 
-// A class to read image data from a particular directory within a tiff file.
-class TiffPixelReader
+// There could be an arbitrary number of samples, but currently we only support 4: red, green, blue, alpha
+#define MAX_SAMPLES 4
+
+enum vcTiffPhotometric
 {
-public:
+  vcTP_GreyScale_WhiteIsZero = 0,
+  vcTP_GreyScale_BlackIsZero = 1,
+  vcTP_RGB                   = 2,
+  vcTP_ColourPalatte         = 3,
+};
 
-  TiffPixelReader()
-    : m_pTiff(nullptr)
-    , m_pointResolution(0.01)
-    , m_origin{}
-    , m_convertFlags(udCCIF_None)
+struct vcTiffSampleData
+{
+  union
   {
+    int64_t   t_int64[MAX_SAMPLES];  // Used to store 1, 2, 4, 8, 16, 32 and 64-bit int samples
+    uint64_t  t_uint64[MAX_SAMPLES]; // Used to store 1, 2, 4, 8, 16, 32 and 64-bit uint samples
+    double    t_double[MAX_SAMPLES]; // float32 values are promoted to doubles
+  };
+};
 
-  }
+struct vcTiffPixelData
+{
+  udDouble3 point;
+  vcTiffSampleData sample;
+};
+
+struct vcTiffImageFormat
+{
+  int32 imageDepth;
+  uint16 bitsPerSample;
+  uint16 sampleFormat;
+  uint16 samplesPerPixel;
+  uint16 photometric;
+  uint16 planarconfig;
+  uint16 **ppColourPalette; // Stored RRR...GGG...BBB..., number of colours == 2 ^ bitsPerSample, also samplesPerPixel MUST = 1
+};
+
+struct vcTiffConvertData;
+
+typedef void(*vcTiffImageReaderDestroyFn)(vcTiffConvertData *pData);
+typedef udError(*vcTiffImageReaderReadPixelFn)(vcTiffConvertData *pData, vcTiffPixelData *pPixelData);
+
+struct TileData
+{
+  uint8_t *pRaster;
+  uint64_t currentPixel;
+  udVector2<uint32> tileDimensions;
+  udVector2<uint32_t> currentTileCoords;
+  uint64_t nTilePixels; // Can be less then the size of a tile
+};
+
+struct ScanLineData
+{
+  uint8_t *pRaster;
+  uint64_t currentPixel;
+  uint32 row;
+};
+
+struct vcTiffConvertData
+{
+  TIFF *pTif;
```

Comment:
Doesn't make a huge amount of sense to rename this from `pTiff`, it results in a number of lines changing and is kind of less clear. Is a TIF different to a TIFF?